src/main/java/org/jfree/chart/date/SerialDate.java

```diff
@@ -423,8 +432,10 @@ public static int stringToMonthCode(String s) {
                 }
             }
         }
+        @SuppressWarnings({"index", "value"}) // the loop above only ensures that 1 <= result <= 12 if s is an actual month name
+        @IntRange(from = 1, to =12) int toReturn = result;
```

Comment:
Here and elsewhere, write an assertion instead of introducing a new temporary variable.